### PR TITLE
Add Razer BlackWidow Essential

### PR DESCRIPTION
**Pull request by fperetti**
This PR should add basic support for Razer Blackwidow Essential keyboard which is similar to Blackwidow 2019 but without RGB led and less functionality.

### CODE REVIEW

File: driver/razerkbd_driver.c [hunk 1]
```diff
@@ -461,6 +461,9 @@ static ssize_t razer_attr_read_device_type(struct device *dev, struct device_att
     case USB_DEVICE_ID_RAZER_BLACKWIDOW_2019:
         device_type = "Razer BlackWidow 2019\n";
         break;
+    case USB_DEVICE_ID_RAZER_BLACKWIDOW_ESSENTIAL:
```
Comment (z3ntu) [inline driver/razerkbd_driver.c:480]: newline before that

File: daemon/openrazer_daemon/hardware/keyboards.py [hunk 2]
```diff
@@ -812,6 +812,24 @@ class RazerBlackWidow2019(_RippleKeyboard):
...
+    USB_VID = 0x1532
+    USB_PID = 0x0237
+    HAS_MATRIX = True
+    MATRIX_DIMS = [6, 22]
+    METHODS = ['get_device_type_keyboard', 'set_static_effect', 'set_none_effect',
+               'set_breath_single_effect', 'set_key_row', 'get_game_mode',
+               'set_game_mode', 'get_macro_mode', 'set_macro_mode', 'get_macro_effect',
+               'set_macro_effect', 'get_macros', 'delete_macro', 'add_macro']
+
+    DEVICE_IMAGE = "https://assets.razerzone.com/eeimages/support/products/1501/1501-blackwidow2019.png"
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/keyboards.py:856]: Please find a picture for your device on https://support.razer.com (or if not possible, somewhere else but hosted by Razer)

**Review by z3ntu — CHANGES_REQUESTED** [review]
Hi, thanks for opening the PR!

Could you please send me the Model & Product number of the keyboard (should be printed on a sticker).

Also please run `./scripts/generate_appstream_file.sh` to add your device to the appstream file.

**Comment by fperetti** [comment]
> Hi, thanks for opening the PR!
> 
> Could you please send me the Model & Product number of the keyboard (should be printed on a sticker).

Hi! sure, this is the model and product numbers from the back of the keyboard:

MODEL: BLACKWIDOW ESSENTIAL
MODEL NO: RZ03-0263
PRODUCT NO: RZ03-02630200-RUZ1

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (fperetti) [reply daemon/openrazer_daemon/hardware/keyboards.py:856]: I can't find the keyboard on razer's site.. neither on any razer's domains, I just keep getting images from others stores, any other suggestions?

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py:856]: Hmm... Maybe find a keyboard which looks more or less like this one and use the picture of that? And add a comment, that there aren't any pictures of this device available or sth

**Comment by z3ntu** [comment]
Thanks!

Did you try all effects and do they work properly?

**Comment by z3ntu** [comment]
Please also add your device to the README.md file.

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (fperetti) [reply daemon/openrazer_daemon/hardware/keyboards.py:856]: Ok, after digging a little bit, the blackwidow-2019 is the most similar keyboard, the main differences are the RGB lights, in the Essential one, it's only green also it's has less lights effects. I'll add a comment then. Thanks!

File: README.md [hunk 3]
```diff
@@ -50,6 +50,7 @@ The below devices are fully feature supported by OpenRazer which means all avail
 | Razer Blade 15 (2018) Mercury                 |  1532:0240  |
 | Razer BlackWidow (2019)                       |  1532:0241  |
 | Razer Blade 15 (Mid 2019) Mercury             |  1532:0245  |
+| Razer BlackWidow Essential                    |  1532:0237  |
```
Comment (z3ntu) [inline README.md]: Please move the line above the `Razer Blade Stealth (2019)` line, so it's sorted by USB PID

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py:856]: this is still missing

**Comment by z3ntu** [comment]
Did you try all effects and do they work properly?

**Comment by seanmuth** [comment]
Trying to bump this PR up, haven't had a chance to try this out with my blackwidow essential, hoping @fperetti can chime in before I try pulling this down and giving it a test drive.

E: I don't mind compiling this from @fperetti's branch, but not seeing any instructions on how to do that in the wiki. tried just running `make` and `scripts/build_debs.sh` but it doesn't seem to be including the changes for this device. Any tips @z3ntu? C isn't my language, I'm a python guy so I can kludge my way getting this to build with a little guidance.

**Comment by z3ntu** [comment]
@seanmuth You should be able to follow https://github.com/openrazer/openrazer/wiki/Building-a-package#ubuntu--linux-mint , just replace the first command with a clone of this PR (if you need help just ask) - you have to run the command from the source tree you want packaged, so you have to check out the correct branch.

**Comment by brinman2002** [comment]
@z3ntu what are you looking for, for testing evidence on this?  I compiled @fperetti 's branch and the driver loads.  This is about the testing I've done (other than I'm typing on this keyboard right now)

```
root@waluigi:/sys/bus/hid/drivers/razerkbd# cat module/version 
2.5.0
root@waluigi:/sys/bus/hid/drivers/razerkbd# ls
0003:1532:0237.0001  0003:1532:0237.0003  module  uevent
0003:1532:0237.0002  bind                 new_id  unbind
```

**Comment by brinman2002** [comment]
@z3ntu I ran Polychromatic Controller and confirmed that I could set the lighting effects to static, as well as adjust the brightness.  The gaming mode toggle appears to work as well.

File: daemon/openrazer_daemon/hardware/keyboards.py (hunk 2 above)
Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/keyboards.py:856]: I found the device on the Chinese Razer store :thinking: http://cn.razerzone.com/gaming-keyboards/razer-blackwidow-essential Doesn't have the pictures I'm looking for though. Let's keep it with that for now then.

**Review by z3ntu — APPROVED** [review]
LGTM, thanks for the PR @fperetti , thanks for testing @brinman2002 !

**Pull request merged**